contributing/giving-constructive-criticism.rst

```diff
@@ -0,0 +1,154 @@
+Consider the amount of work and time the author has spend on there idea.
+And how your response would make them feel;
+
+* Are you making a point only because you believe they are wrong?
+
+* Did you correctly understand there intention? or are you making assumptions?
+
+Maybe you can ask for clarification?, suggest an alternative? or
+provide a simple explanation *why* disagree with there proposal.
+
+Even if something *is* really wrong or "a bad idea", stay respectful and
+don't get into endless you-are-wrong discussions or "flame wars".
+
+* Good questions avoid judgment and avoid assumptions about the author's
+  perspective.
+
+* Avoid using terms that could be seen as referring to personal traits. ("dumb",
+  "stupid"). Assume everyone is intelligent and well-meaning.
+
+* Be explicit. Remember people don't always understand your intentions online.
+
+* Don't use hyperbole. ("always", "never", "endlessly", "nothing", "worst",
+  "horrible", "terrible").
+
+Instead of saying "This is a horrible idea" use something like
+"This can introduce issues in the future. Please make sure this is safe to use."
+
+Asking for changes
+------------------
+
+Rarely something is perfect from the start, while the code itself is good.
+It may not be optimal or conform the Symfony coding style.
+
+Again, understand the author already spend time on the issue and asking
+for (small) changes may be misinterpreted or seen as a personal attack.
+
+Be thankful for there work (so far), stay positive and really help them
+to make the contribution a great one. *Especially if they are a first
+time contributor.*
+
+Use words like "Please", "Thank you" and "Could you" instead of making demands;
+
+* "Thank you for your work so far. I left some suggestions for improvement
+  to make the code more readable."
+
+* "Your code contains some coding-style problems, can you fix these before
+  we merge? Thank you"
+
+* "Please use 4 spaces instead of tabs", "This needs be on the previous line";
+
+  During a pull request review you can usually leave more then one comment,
+  you don't have to use "Please" all the time. But it wouldn't hurt.
+
+It may not seem like much, but saying "Thank you" does make others feel
+more welcome.
+
+Making jokes (using humor)
+--------------------------
+
+In short: Don't be a troll; This violates the Code of Conduct and may
+even get you banned!
+
+OK, lets be honest. We all make jokes, some are funny and some are not.
+They don't put anyone of, right? Well, who's to tell. What's funny to one
+is offensive to another.
+
+In short. Keep it real. There is nothing wrong with using a meme or emoji
+from time to time, but don't over do it. And even then, keep it friendly.
+
+**Don't use sarcasm for a serious topic, that's not something that belongs
+to the Symfony community.** And don't marginalize someone's problems;
+"Well I guess that's not suppose to happen? 😆 ".
+
+Even if someone's explanation is "inviting to joke about", it's a real
```

Comment:
the quoted sentence doesn't parse well here, maybe:

`... explanation is "inviting you to joke about it"...`?